packages/shaders/src/shader-mount.ts

```diff
@@ -136,34 +149,103 @@ export class ShaderMount {
     this.uniformLocations = uniformLocations;
   };
 
+  /**
+   * The scale that we should render at.
+   * - Used to target 2x rendering even on 1x screens for better antialiasing
+   * - Prevents the virtual resolution from going beyond the maximum resolution
+   * - Accounts for the page zoom level so we render in physical device pixels rather than CSS pixels
+   */
+  private renderScale = 1;
+  private parentWidth = 0;
+  private parentHeight = 0;
+
   private resizeObserver: ResizeObserver | null = null;
   private setupResizeObserver = () => {
-    this.resizeObserver = new ResizeObserver(() => this.handleResize());
+    this.resizeObserver = new ResizeObserver(([entry]) => {
+      if (entry?.borderBoxSize[0]) {
+        this.parentWidth = entry.borderBoxSize[0].inlineSize;
+        this.parentHeight = entry.borderBoxSize[0].blockSize;
+      }
+
+      this.handleResize();
+    });
+
     this.resizeObserver.observe(this.parentElement);
+    visualViewport?.addEventListener('resize', this.handleVisualViewportChange);
+
+    const rect = this.parentElement.getBoundingClientRect();
+    this.parentWidth = rect.width;
+    this.parentHeight = rect.height;
```

Comment:
- Using client width/height was resulting in that size twitching during zoom
- The ResizeObserver gives us precise measurements anyway